libc/src/setjmp/arm/setjmp.cpp

```diff
@@ -0,0 +1,59 @@
+//===-- Implementation of setjmp ------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/__support/common.h"
+#include "src/setjmp/setjmp_impl.h"
+
+namespace LIBC_NAMESPACE {
+
+#if defined(__thumb__) && __ARM_ARCH_ISA_THUMB == 1
+
+[[gnu::naked, gnu::target("thumb")]]
```

Comment:
just curious, does `arm/aarch64` also support multi-versioned symbols?

Reply:
If multi-versioned symbols is referring to function multi-versioning then there is a beta version defined in https://arm-software.github.io/acle/main/acle.html#function-multi-versioning AFAIK it has been concentrating on AArch64 though.

Unlikely to be used in bare-metal cases like cortex-m0 though.

Reply:
Right, depends on loader support, which we currently lack.  Also, not sure yet we _want_ to support a libc with mixed arm/thumb mode support.  Frankly, I'd suspect that folks that _care_ about libc support on 32b ARM only really care about Thumb mode (unless they're running on some variant with only ARM mode support).